### PR TITLE
fix `as_result_df` for nested `rbind`

### LINKED ISSUE #815: `as_result_df()` fails with `names` error
Apologies for the longish reproducible example.  This seems like a regression on `as_result_df()` behavior, as previously this code was working fine (at least I suppose so, we were using `g_forest()`)

```r
library(tern)
library(dplyr)

AGE_probs <- 0.5

adtte <- scda::synthetic_cdisc_data("rcd_2022_06_27")$adtte %>%
  df_explicit_na() %>%
  filter(
    PARAMCD == "OS"
  ) %>%
  mutate(
    AVAL = day2month(AVAL),
    AVALU = "Months",
    is_event = CNSR == 0,
    ARM_BIN = fct_collapse_only(
      ARM,
      CTRL = "B: Placebo",
      TRT = c("A: Drug X", "C: Combination")
    ),
    AGE_BIN = cut_quantile_bins(AGE, probs = AGE_probs),
    BMRKR2 = fct_explicit_na_if(BMRKR2, BEP01FL == "N"),
    BMRKR2_BIN = fct_collapse_only(
      BMRKR2,
      High = "HIGH",
      Low = c("MEDIUM", "LOW")
    )
  ) %>%
  var_relabel(
    BEP01FL = "BEP",
    BMRKR2_BIN = "Biomarker (Binary)"
  )
tables_all <- function(filter_var, filter_condition, sub_var) {
  dataset <- adtte %>%
    filter(!!as.name(filter_var) == filter_condition)
  if (nrow(dataset) == 0) {
    stop(paste("Subset", filter_var, "==", filter_condition, "is empty"))
  }
  tbl <- extract_survival_subgroups(
    variables = list(
      tte = "AVAL",
      is_event = "is_event",
      arm = "ARM_BIN",
      subgroups = sub_var
    ),
    label_all = "ITT",
    data = dataset
  )
  basic_table() %>%
    tabulate_survival_subgroups(
      df = tbl,
      vars = c("n_tot_events", "n", "n_events", "median", "hr", "ci"),
      time_unit = dataset$AVALU[1]
    )
}
add_subtitle <- function(sub_tab, sub_title) {
  label_at_path(sub_tab, path = row_paths(sub_tab)[[1]][1]) <- sub_title
  sub_tab
}
tables_list <- list(
  tables_all(filter_var = "BMRKR2_BIN", filter_condition = "High", sub_var = "SEX"),
  tables_all(filter_var = "BMRKR2_BIN", filter_condition = "Low", sub_var = "SEX"),
  tables_all(filter_var = "BMRKR2_BIN", filter_condition = "High", sub_var = "AGE_BIN"),
  tables_all(filter_var = "BMRKR2_BIN", filter_condition = "Low", sub_var = "AGE_BIN")
)
col_info(tables_list[[2]]) <- col_info(tables_list[[1]])
result <- rbind(
  add_subtitle(tables_list[[1]], "BMRKR2_BIN = High"),
  add_subtitle(tables_list[[2]], "BMRKR2_BIN = Low"),
  add_subtitle(tables_list[[3]], "BMRKR2_BIN = High"),
  add_subtitle(tables_list[[4]], "BMRKR2_BIN = Low")
)
result
as_result_df(result)
```

Fails with:
```
Error in names(object) <- nm : 
  'names' attribute [13] must be the same length as the vector [12]
```

Session info:
```
> utils::sessionInfo()
R version 4.3.2 (2023-10-31)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS Ventura 13.4

Matrix products: default
BLAS:   /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Zurich
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] dplyr_1.1.4           tern_0.9.3.9007       rtables_0.6.6.9003   
[4] magrittr_2.0.3        formatters_0.5.5.9004

loaded via a namespace (and not attached):
 [1] Matrix_1.6-4      gtable_0.3.4      compiler_4.3.2    tidyselect_1.2.0 
 [5] tidyr_1.3.0       splines_4.3.2     scales_1.3.0      fastmap_1.1.1    
 [9] lattice_0.22-5    ggplot2_3.4.4     R6_2.5.1          generics_0.1.3   
[13] forcats_1.0.0     rbibutils_2.2.16  backports_1.4.1   checkmate_2.3.1  
[17] tibble_3.2.1      munsell_0.5.0     scda_0.1.6.9014   pillar_1.9.0     
[21] rlang_1.1.3       utf8_1.2.4        stringi_1.8.3     broom_1.0.5      
[25] cli_3.6.2         Rdpack_2.6        digest_0.6.33     grid_4.3.2       
[29] rstudioapi_0.15.0 lifecycle_1.0.4   vctrs_0.6.5       glue_1.7.0       
[33] survival_3.5-7    fansi_1.0.6       colorspace_2.1-0  purrr_1.0.2      
[37] tools_4.3.2       pkgconfig_2.0.3   htmltools_0.5.7 
```

### PR DESCRIPTION
**Pull request by Melkiades**
Fixes #815 and #809

Having multiple nested `rbind` is usually a problem as it breaks the structure of the table completely. In other words, we should find a way to flatten `rbind_roots` nodes, or remove them in {tern} (`tabulate_survival_subgroups` uses them), or establish a way to represent these in the table tree.

I would suggest going for the following:

- [ ] Harmonize flattening of rbinds with proper representation of the "split" with at least its visible label rows as split values (different representation)
- [ ] Remove rbinds from {tern} when not from different data (I think the example case I got could be done with a split as the filtering variable can be the split) https://github.com/insightsengineering/tern/issues/1181
- [x] Add LabelRows to the result df (probably with another result_df_* function)

### CODE REVIEW

File: tests/testthat/test-result_data_frame.R [hunk 1]
```diff
@@ -119,13 +119,104 @@ test_that("as_result_df works with visual output (as_viewer)", {
...
+    df_to_tt(ard_out) %>% row.names()
+  )
+  
+  init_tbl <- df_to_tt(mtcars) 
+  end_tbl <- init_tbl %>% as_result_df(as_is = TRUE) %>% df_to_tt()
+  
+  expect_equal(
+    matrix_form(init_tbl)$strings, 
+    matrix_form(end_tbl)$strings
+  )
```
Comment (Melkiades) [inline tests/testthat/test-result_data_frame.R:221]: hey @shajoezhu could you check this? Tell me if it is as you want it. The above example works when there is a complex table and non-unique rownames

Comment (shajoezhu) [reply tests/testthat/test-result_data_frame.R:221]: brilliant! i was wondering why it has non-unique rownames (what about appending the multi-level rownames, and make them unique?)

Comment (Melkiades) [reply tests/testthat/test-result_data_frame.R:221]: rownames need to be unique but if you create a table using {rtables} it happens that they are not unique. If that is the case as_result_df makes a label_name column with the final row names and if that column is present df_to_tt uses it for the table

Comment (shajoezhu) [reply tests/testthat/test-result_data_frame.R:221]: i would suggest we take look into the tree structure, i feel that we need to go deeper, and adding more elements at the node level

if I do the following, i see it starts to appending new columsn

```
> df_to_tt(mtcars) %>% as_result_df()
             avar_name            row_name          label_name row_num is_group_summary node_class  mpg cyl  disp  hp drat    wt  qsec vs am gear carb
1            Mazda RX4           Mazda RX4           Mazda RX4       1            FALSE    DataRow   21   6   160 110  3.9  2.62 16.46  0  1    4    4
2        Mazda RX4 Wag       Mazda RX4 Wag       Mazda RX4 Wag       2            FALSE    DataRow   21   6   160 110  3.9 2.875 17.02  0  1    4    4
3           Datsun 710          Datsun 710          Datsun 710       3            FALSE    DataRow 22.8   4   108  93 3.85  2.32 18.61  1  1    4    1
4       Hornet 4 Drive      Hornet 4 Drive      Hornet 4 Drive       4            FALSE    DataRow 21.4   6   258 110 3.08 3.215 19.44  1  0    3    1
5    Hornet Sportabout   Hornet Sportabout   Hornet Sportabout       5            FALSE    DataRow 18.7   8   360 175 3.15  3.44 17.02  0  0    3    2
6              Valiant             Valiant             Valiant       6            FALSE    DataRow 18.1   6   225 105 2.76  3.46 20.22  1  0    3    1
7           Duster 360          Duster 360          Duster 360       7            FALSE    DataRow 14.3   8   360 245 3.21  3.57 15.84  0  0    3    4
8            Merc 240D           Merc 240D           Merc 240D       8            FALSE    DataRow 24.4   4 146.7  62 3.69  3.19    20  1  0    4    2
9             Merc 230            Merc 230            Merc 230       9            FALSE    DataRow 22.8   4 140.8  95 3.92  3.15  22.9  1  0    4    2
10            Merc 280            Merc 280            Merc 280      10            FALSE    DataRow 19.2   6 167.6 123 3.92  3.44  18.3  1  0    4    4
11           Merc 280C           Merc 280C           Merc 280C      11            FALSE    DataRow 17.8   6 167.6 123 3.92  3.44  18.9  1  0    4    4
12          Merc 450SE          Merc 450SE          Merc 450SE      12            FALSE    DataRow 16.4   8 275.8 180 3.07  4.07  17.4  0  0    3    3
13          Merc 450SL          Merc 450SL          Merc 450SL      13            FALSE    DataRow 17.3   8 275.8 180 3.07  3.73  17.6  0  0    3    3
14         Merc 450SLC         Merc 450SLC         Merc 450SLC      14            FALSE    DataRow 15.2   8 275.8 180 3.07  3.78    18  0  0    3    3
15  Cadillac Fleetwood  Cadillac Fleetwood  Cadillac Fleetwood      15            FALSE    DataRow 10.4   8   472 205 2.93  5.25 17.98  0  0    3    4
16 Lincoln Continental Lincoln Continental Lincoln Continental      16            FALSE    DataRow 10.4   8   460 215    3 5.424 17.82  0  0    3    4
17   Chrysler Imperial   Chrysler Imperial   Chrysler Imperial      17            FALSE    DataRow 14.7   8   440 230 3.23 5.345 17.42  0  0    3    4
18            Fiat 128            Fiat 128            Fiat 128      18            FALSE    DataRow 32.4   4  78.7  66 4.08   2.2 19.47  1  1    4    1
19         Honda Civic         Honda Civic         Honda Civic      19            FALSE    DataRow 30.4   4  75.7  52 4.93 1.615 18.52  1  1    4    2
20      Toyota Corolla      Toyota Corolla      Toyota Corolla      20            FALSE    DataRow 33.9   4  71.1  65 4.22 1.835  19.9  1  1    4    1
21       Toyota Corona       Toyota Corona       Toyota Corona      21            FALSE    DataRow 21.5   4 120.1  97  3.7 2.465 20.01  1  0    3    1
22    Dodge Challenger    Dodge Challenger    Dodge Challenger      22            FALSE    DataRow 15.5   8   318 150 2.76  3.52 16.87  0  0    3    2
23         AMC Javelin         AMC Javelin         AMC Javelin      23            FALSE    DataRow 15.2   8   304 150 3.15 3.435  17.3  0  0    3    2
24          Camaro Z28          Camaro Z28          Camaro Z28      24            FALSE    DataRow 13.3   8   350 245 3.73  3.84 15.41  0  0    3    4
25    Pontiac Firebird    Pontiac Firebird    Pontiac Firebird      25            FALSE    DataRow 19.2   8   400 175 3.08 3.845 17.05  0  0    3    2
26           Fiat X1-9           Fiat X1-9           Fiat X1-9      26            FALSE    DataRow 27.3   4    79  66 4.08 1.935  18.9  1  1    4    1
27       Porsche 914-2       Porsche 914-2       Porsche 914-2      27            FALSE    DataRow   26   4 120.3  91 4.43  2.14  16.7  0  1    5    2
28        Lotus Europa        Lotus Europa        Lotus Europa      28            FALSE    DataRow 30.4   4  95.1 113 3.77 1.513  16.9  1  1    5    2
29      Ford Pantera L      Ford Pantera L      Ford Pantera L      29            FALSE    DataRow 15.8   8   351 264 4.22  3.17  14.5  0  1    5    4
30        Ferrari Dino        Ferrari Dino        Ferrari Dino      30            FALSE    DataRow 19.7   6   145 175 3.62  2.77  15.5  0  1    5    6
31       Maserati Bora       Maserati Bora       Maserati Bora      31            FALSE    DataRow   15   8   301 335 3.54  3.57  14.6  0  1    5    8
32          Volvo 142E          Volvo 142E          Volvo 142E      32            FALSE    DataRow 21.4   4   121 109 4.11  2.78  18.6  1  1    4    2
```

however, if i start appending to this operation, it does not go back

```
 df_to_tt(mtcars) %>% as_result_df() %>% df_to_tt()  %>% as_result_df()
```

**Review by shajoezhu — APPROVED** [review]
lgtm! thanks @Melkiades

**Pull request merged**